webrtc-c/canary/src/Cloudwatch.cpp

```diff
@@ -0,0 +1,81 @@
+#include "Include.h"
+
+namespace Canary {
+
+Cloudwatch::Cloudwatch(Canary::PConfig pConfig, ClientConfiguration* pClientConfig)
+    : logs(pConfig, pClientConfig), monitoring(pConfig, pClientConfig), terminated(FALSE)
+{
+}
+
+STATUS Cloudwatch::init(Canary::PConfig pConfig)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    ClientConfiguration clientConfig;
+    CreateLogGroupRequest createLogGroupRequest;
+    Aws::CloudWatchLogs::Model::CreateLogStreamOutcome createLogStreamOutcome;
+    CreateLogStreamRequest createLogStreamRequest;
+
+    clientConfig.region = pConfig->pRegion;
+    auto& instance = getInstanceImpl(pConfig, &clientConfig);
+
+    if (STATUS_FAILED(instance.logs.init())) {
+        DLOGW("Failed to create Cloudwatch logger, fallback to local output");
```

Comment:
I think our fallback should be file logger like we do producer-c. This is because, we do want to preserve logs in files to pull out. The apps we mostly be run as background on CI/CD and we might not have a backup.

Reply:
@disa6302 since we run this on Jenkins, we can simply get the log by fetching `${BUILD_URL}/consoleText`. I'm trying to keep our Canary as simple as possible, so that we can make it really robust for our alarms, I don't want to include things that can be done from outside.

Even if the user doesn't use Jenkins, they can easily just use `tee` to store the logs to a file.

Reply:
Doesnt `tee` just store what is on the console? What about when the logs run for days? 
Also, if we run the application as a background process, the logs would not appear on the Jenkins console as well. Would ${BUILD_URL}/consoleTest store logs that do not show up on the Jenkins console?

IMO, adding a fileLogger option is not going to add any complexity. We can skip it for now, but, I think we should not rely on external mechanisms for logging.

Reply:
In that case, we can just use `split` (it allows you to break up big input to chunks, you can even set the size limit of each chunk):

![Screen Shot 2020-08-13 at 1 05 08 PM](https://user-images.githubusercontent.com/15654932/90181691-dce78680-dd7e-11ea-898e-53eabfa14b5a.png)

Why would we run the application as a background process? Why can't we just spawn a new job and run the application in that job?

Reply:
Discussed offline with @disa6302 , we made a decision to add the file logger.

Reply:
Added file logger.